### PR TITLE
Support to config custom decryption methods for kafka configurations and fix bug kafka-reporter-plugin exception "java.lang.NoClassDefFoundError: com/google/gson/Gson"`

### PR DESCRIPTION
**Pull request by GuiSong01**
[Bug] fix bug `kafka-reporter-plugin-8.15.0 throw exception "java.lang.NoClassDefFoundError: com/google/gson/Gson"`

### Fix <https://github.com/apache/skywalking/issues/10848>
- [ ] Add a unit test to verify that the fix works.
- [ ] Explain briefly why the bug exists and how to fix it.

### [skywalking-java support to decrypt password of kafka with SASL_SSL enabled](https://github.com/apache/skywalking/issues/10815)
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [x] Update the documentation to include this new feature.
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.

- [x]  Closes #<10848>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ Release Notes.
 * Change the classloader to locate the agent path in AgentPackagePath, from `SystemClassLoader` to AgentPackagePath's loader.
 * Support Grizzly Trace
 * Fix possible IllegalStateException when using Micrometer.
-
+* Support to config custom decryption methods for kafka configurations and fix bug `kafka-reporter-plugin-8.15.0 throw exception "java.lang.NoClassDefFoundError: com/google/gson/Gson"`
```
Comment (wu-sheng) [inline CHANGES.md:20]: This should be two features, rather than one.

File: docs/en/setup/service-agent/java-agent/configurations.md [hunk 2]
```diff
@@ -103,6 +103,8 @@ This is the properties list supported in `agent/config/agent.config`.
...
+| `plugin.kafka.decrypt_class`                                    | Specify which class to decrypt encrypted configuration of kafka.You can set encrypted information in `plugin.kafka.producer_config_json` or `plugin.kafka.producer_config` if you need.                                                                                                                                                                                                                                                                                                                                                                | SW_KAFKA_DECRYPT_CLASS                                           | ``                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/configurations.md:106]: https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/advanced-reporters/

This doc should be updated as well, including
1. Which artifact and version should the developer depend on to develop this? Which interface should be implemented?
2. Where to put the packaged codes

File: apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaProducerManager.java [hunk 3]
```diff
@@ -153,6 +145,18 @@ private void notifyListeners(KafkaConnectionStatus status) {
...
 
+    private Map<String, String> decrypt(Map<String, String> config) {
+        try {
+            Class<?> decryptClazz = Class.forName(Kafka.DECRYPT_CLASS);
+            Method decryptMethod = decryptClazz.getMethod(Kafka.DECRYPT_METHOD, Map.class);
+            return (Map<String, String>) decryptMethod.invoke(decryptClazz.newInstance(), config);
+        } catch (ClassNotFoundException | NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException e) {
+            // ignore
+            LOGGER.warn("The decrypt class {} is not exist, exception:{}.", Kafka.DECRYPT_CLASS, e);
+            return config;
```
Comment (wu-sheng) [inline apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaProducerManager.java:156]: If you want to do in this way, please indicate in the docs, what method should be written.

**Comment by wu-sheng** [comment]
>  Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.2.1:check (validate) on project kafka-reporter-plugin: You have 2 Checkstyle violations. -> [Help 1]

You should do code style check.

**Comment by wu-sheng** [comment]
As this PR is not updated, GSON part is being fixed by https://github.com/apache/skywalking-java/pull/538 as well.

**Comment by GuiSong01** [comment]
> > Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.2.1:check (validate) on project kafka-reporter-plugin: You have 2 Checkstyle violations. -> [Help 1]
> 
> You should do code style check.

**Pull request closed without merging**